rules/jsll/src/rules/jsll-awa-init/jsll-awa-init.ts

```diff
@@ -38,73 +40,91 @@ const rule: IRuleBuilder = {
                         }
 
                         isFirstExpressionStatement = false;
-
                         validateAwaInit(node, eslintContext, true);
-
                         validated = true;
-                    },
-                    'Program:exit'(programNode) {
-                        if (hasJSLLScript && (!validated)) {
-                            // Should verify init but no ExpressionStatement was encountered.
-                            // e.g.:
-                            // ...
-                            // <script src="../jsll-4.js"></script>
-                            // <script>var a = 2;</script>
-                            eslintContext.report(programNode, `JSLL is not initialized with "awa.init(config)" function. Initialization script should be placed immediately after JSLL script.`);
-                        }
                     }
                 };
             }
         });
 
         const validateScript = async (scriptParse: IScriptParse) => {
-            if (!hasJSLLScript || !isHead) {
+            const sourceCode = scriptParse.sourceCode;
+
+            if (!isPotentialInitScript(sourceCode)) {
+                immediateAfterJSLLscript = false;
+
+                return;
+            }
+
+            if (!traverseStarts) {
+                // When jsll init script is included in an external script
+                // `parse::javascript` of the init script is emitted before `element::script` of the JSLL link.
```

Comment:
Should we try to change this behavior? If not maybe we should document it somewhere in the main repo?

Reply:
This is due to `parseJavascript` subscribes to `fetch::end`, but `parseJavaScriptTag` subscribes to `element::script`. I'm afraid changing this behavior will be difficult. I think documenting the sequence of the events emitted will be very helpful.
```
        sonarwhal.on('fetch::end', this.parseJavascript.bind(this));
        sonarwhal.on('element::script', this.parseJavascriptTag.bind(this));
```